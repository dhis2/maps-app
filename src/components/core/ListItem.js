import React from 'react';
import PropTypes from 'prop-types';
import styles from './styles/ListItem.module.css';

// Left aligned label (no wrap) and right align value of variable width
const ListItem = ({ label, formatter, children }) =>
    children !== null && children !== undefined && children !== '' ? (
        <table className={styles.listItem}>
            <tbody>
                <tr>
                    <th>{label}</th>
                    <td>{formatter ? formatter(children) : children}</td>
                </tr>
            </tbody>
        </table>
    ) : null;

ListItem.propTypes = {
    label: PropTypes.string.isRequired,
    formatter: PropTypes.func,
    children: PropTypes.oneOfType([
        PropTypes.arrayOf(PropTypes.node),
        PropTypes.node,
    ]),
};

export default ListItem;
