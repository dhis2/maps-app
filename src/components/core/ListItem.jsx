import PropTypes from 'prop-types'
import React from 'react'
import styles from './styles/ListItem.module.css'

// Left aligned label (no wrap) and right align value of variable width
const ListItem = ({ label, formatter, children }) =>
    children !== null && children !== undefined && children !== '' ? (
        <table className={styles.listItem}>
            <tbody>
                <tr>
                    <th>{label}</th>
                    <td>{formatter ? formatter(children) : children}</td>
                </tr>
            </tbody>
        </table>
    ) : null

ListItem.propTypes = {
    label: PropTypes.string.isRequired,
    children: PropTypes.oneOfType([
        PropTypes.arrayOf(PropTypes.node),
        PropTypes.node,
    ]),
    formatter: PropTypes.func,
}

export default ListItem
