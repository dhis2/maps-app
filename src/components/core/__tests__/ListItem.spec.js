import React from 'react';
import { shallow } from 'enzyme';
import ListItem from '../ListItem';

describe('Drawer', () => {
    it('should not render if children is missing', () => {
        const wrapper = shallow(<ListItem label="My label" />);

        expect(wrapper.type()).toBe(null);
    });

    it('should render if a value is passed as children', () => {
        const wrapper = shallow(<ListItem label="My label">123</ListItem>);

        expect(wrapper.type()).toEqual('table');
        expect(wrapper.find('td').prop('children')).toEqual('123');
    });

    it('should display a label', () => {
        const wrapper = shallow(<ListItem label="My label">123</ListItem>);

        expect(wrapper.find('th').prop('children')).toEqual('My label');
    });

    it('should change value if formatter is passed', () => {
        const formatter = jest.fn(value => ++value);
        const wrapper = shallow(
            <ListItem label="My label" formatter={formatter}>
                123
            </ListItem>
        );

        expect(formatter).toHaveBeenCalledWith('123');
        expect(wrapper.find('td').prop('children')).toEqual(124);
    });
});
