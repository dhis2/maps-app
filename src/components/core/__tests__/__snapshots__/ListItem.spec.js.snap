// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer should change value if formatter is passed 1`] = `
<div>
  <table
    class="listItem"
  >
    <tbody>
      <tr>
        <th>
          Item 1 in list
        </th>
        <td>
          124
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Drawer should not render if children is missing 1`] = `<div />`;

exports[`Drawer should render if a value is passed as children 1`] = `
<div>
  <table
    class="listItem"
  >
    <tbody>
      <tr>
        <th>
          Item 1 in list
        </th>
        <td>
          123
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
